contracts/tokenomics/generator/src/contract.rs

```diff
@@ -283,6 +315,86 @@ pub fn execute(
     }
 }
 
+/// Add or remove tokens to and from the blocked list. Returns a [`ContractError`] on failure.
+fn update_tokens_blockedlist(
+    deps: DepsMut,
+    info: MessageInfo,
+    add: Option<Vec<AssetInfo>>,
+    remove: Option<Vec<AssetInfo>>,
+) -> Result<Response, ContractError> {
+    if add.is_none() && remove.is_none() {
+        return Err(ContractError::Std(StdError::generic_err(
+            "Need to provide add or remove parameters",
+        )));
+    }
+
+    let mut cfg = CONFIG.load(deps.storage)?;
+
+    // Permission check
+    if info.sender != cfg.owner {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Remove tokens from blocked list
+    if let Some(asset_infos) = remove {
+        for asset_info in asset_infos {
+            let index = cfg
+                .blocked_list_tokens
+                .iter()
+                .position(|x| *x == asset_info)
+                .ok_or_else(|| {
+                    StdError::generic_err(
+                        "Can't remove token. It is not found in the blocked list.",
+                    )
+                })?;
+            cfg.blocked_list_tokens.remove(index);
+        }
+    }
+
+    // Add tokens to the blocked list
+    if let Some(asset_infos) = add {
+        // ASTRO or Terra native assets (UST, LUNA etc) cannot be blocked
+        let astro = token_asset_info(cfg.astro_token.clone());
+        let uusd = native_asset_info(UUSD_DENOM.to_string());
+        let uluna = native_asset_info(ULUNA_DENOM.to_string());
+
+        if asset_infos.contains(&astro)
+            || asset_infos.contains(&uusd)
+            || asset_infos.contains(&uluna)
```

Comment:
To check other native tokens `is_native_token` method of `AssetInfo` can be used